app/models/user_action.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class UserAction < ApplicationRecord
+  belongs_to :acting_user_verification, class_name: 'UserVerification', optional: true
+  belongs_to :subject_user_verification, class_name: 'UserVerification'
+  belongs_to :user_action_event
+
+  enum :status, {
+    initial: 'initial',
+    success: 'success',
+    error: 'error'
+  }
+
+  validates :status, presence: true, inclusion: { in: statuses.keys }
```

Comment:
Can you put validate on the enum itself? Enum comes with validation built in and will catch the `ArgumentError`

```suggestion
  enum :status, {
    initial: 'initial',
    success: 'success',
    error: 'error'
  }, validate: true
```